GitCommands/Git/GitModule.cs

```diff
@@ -2289,20 +2292,22 @@ public IReadOnlyList<GitStash> GetStashes(bool noLocks = false)
                 diffOptions
             };
 
-            var cache = cacheResult &&
+            CommandCache? cache = cacheResult &&
                         !string.IsNullOrEmpty(secondRevision) &&
                         !string.IsNullOrEmpty(firstRevision) &&
                         !secondRevision.IsArtificial() &&
                         !firstRevision.IsArtificial()
                 ? GitCommandCache
                 : null;
 
-            var patch = _gitExecutable.GetOutput(
+            bool nonZeroGitExitCode = firstId == ObjectId.WorkTreeId && secondId is not null && !isTracked;
```

Comment:
Should not be here but at the caller (may check arguments).
!isTracked cannot use git-diff (regardless of first/second commit).

Reply:
repro:
- create a new file (untracked)
- select `Working directory`, then `Ctrl`+`click` the checked out commit, which turns the added (exit code 0) into a removed file from this perspective (exit code 1)

Reply:
The caller should not request this information, it is meaningless.

Reply:
We need to have at least a comment explaining this. It is not obvious what this check is and why it is needed.

Reply:
> Should not be here but at the caller (may check arguments).
!isTracked cannot use git-diff (regardless of first/second commit).

> The caller should not request this information, it is meaningless.

Explain it in other words, please.

> It is not obvious what this check is and why it is needed.

Would you better understand it with `nonZeroGitExitCodeExpected`?
It is used once - in the next statement.